src/api/modelTypes.js

```diff
@@ -749,9 +749,42 @@ type MessageBase = $ReadOnly<{|
   content_type: 'text/html',
   // display_recipient handled on PmMessage and StreamMessage separately
 
-  // Optional because we only store it in Redux if it's coming from a server
-  // with FL >=118; see comment on MessageEdit.
-  edit_history: $ReadOnlyArray<MessageEdit> | null,
+  /**
+   * A possibly incomplete view of the message's edit history.
+   *
+   * Null if it's coming from a server with FL <118; see comment on
+   * MessageEdit.
+   *
+   * Missing/undefined if the message had no edit history when we added it
+   * to the state.
+   *
+   * Missing/undefined if, at the time we added it to the state, the realm
+   * didn't allow viewing edit history.
+   *
+   * Stale if the message was updated or moved after we added it to the state.
+   */
+  // TODO: Keep reasonably current:
+  // - Use `null` for the allow-edit-history-false case, to distinguish it
+  //   from the message-never-edited case. (Handle everywhere we add/update
+  //   Messages in Redux: the get-messages response, new-message events,
+  //   update-message events.)
+  // - Handle changes to the allow-edit-history setting:
+  //   - When allow-edit-history is turned off, set all messages'
+  //     edit_history to `null`. (Otherwise we'd be keeping around
+  //     edit_history data that admins would rather we didn't have…and also,
+  //     that data would get irrecoverably stale as messages get updated.)
+  //   - When allow-edit-history is turned on…probably just leave
+  //     edit_history `null` on messages we know about, don't force `null`
+  //     on new or newly fetched messages, and expect a re-/register
+  //     soon?
```

Comment:
As we discussed in yesterday's call, let's simplify this:

We'll plan to treat an allow-edit-history change similar to a `restart` event (in particular, a `restart` event where the server version changed, which could mean any number of subtle changes in server behavior), and do a re-fetch of server data soon after it.

* To track this task, can add it to the list in #3408. Then for now, we don't worry about it because we frequently re-fetch anyway.

Reply:
> can add it to the list in

Done.